core/new-gui/src/app/common/formly/formly-utils.ts

```diff
@@ -18,6 +18,40 @@ export function setHideExpression(toggleHidden: string[], fields: FormlyFieldCon
   });
 }
 
+/* Factory function to make functions that hide expressions for a particular field */
+export function createShouldHideFieldFunc(
+  hideTarget: string,
+  hideType: "regex" | "equals" | string,
```

Comment:
can we restrict the `hideType` by removing the `string`?

Reply:
The place where the function is being called has a string from the backend that we can't assume is "regex" or "equals". But we could check for if the hideType is not regex or equals outside the function, and change the CustomJSONSchema7 interface to reflect that. I'll make that change.